python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -354,15 +354,29 @@ def removeNodeAttribute(self, attrib_name):
         if attrib_name in self._data.keys():
             del self._data[attrib_name]
 
-    def removeNodeIndexCacheAttribute(self, attrib_name):
-        if self._index_cache and attrib_name in self._index_cache.keys():
-            del self._index_cache[attrib_name]
-
     def setNodeIndexCache(self, new_index_cache):
+        if self.fullyLoadedFromDisk():
+            logger.debug("Warning: Setting IndexCache data on live object, please avoid!")
```

Comment:
No, not for now, I don't think we're doing this anywhere out of reading in the index anyway when it's not fully loaded. Once we know we're safe we can consider this but I'm not comfortable doing this for now.